src/puppetlabs/trapperkeeper/services/authorization/authorization_core.clj

```diff
@@ -37,10 +41,12 @@
 (defn- build-rule
   "Build a new Rule based on the provided config-map"
   [config-map]
-  (let [rule-type (keyword (get config-map :type :path))]
-    (if (= rule-type :path)
-      (-> (rules/new-path-rule (config-map :path) (method config-map)))
-      (-> (rules/new-regex-rule (config-map :path) (method config-map))))))
+  (let [rule-type (keyword (get config-map :type :path))
+        rule-fn (rule-type new-rule-func-map)
+        rule (rule-fn (config-map :path) (method config-map))]
```

Comment:
Hm, I find this much harder to follow using a function map.  Considering that we don't expect there to be more than 2 or 3 different branches, I would prefer a straightforward `cond` expression.

> the function has two different behaviors with the condition in place

You could argue it still has different behaviors with a function map, it's just decoupled from the different types of rules that determine the behavior.  
I would consider this as a good thing if we knew we were going to be adding more types of rules and all we'd need to do is add a new function to the map, but I don't think that's the case.
Or if we had multiple places where we wanted to use the function map and reduce duplication.

I know I know, higher-order functions and all, but I think a `cond` is easier in this case.

Not going to make a fuss over it though, just explaining my thinking :smile: It's a fairly isolated piece of code, so not a big deal.

@camlow325 you got any thoughts?